File: engine/src/main/java/com/terralite/engine/chunk/ChunkStorage.java
package com.terralite.engine.chunk;

import java.util.Collection;
import java.util.Optional;

public interface ChunkStorage {
    Optional<Chunk> get(ChunkPos pos);

    boolean contains(ChunkPos pos);

    Chunk put(Chunk chunk);

    Chunk remove(ChunkPos pos);

    Chunk require(ChunkPos pos);

    Collection<ChunkPos> positions();

    Collection<Chunk> chunks();

    int size();
}

File: engine/src/commonMain/kotlin/com/terralite/engine/chunk/ChunkStorage.kt
package com.terralite.engine.chunk

interface ChunkStorage {
    fun get(pos: ChunkPos): Chunk?
    fun contains(pos: ChunkPos): Boolean
    fun put(chunk: Chunk): Chunk
    fun remove(pos: ChunkPos): Chunk
    fun require(pos: ChunkPos): Chunk
    fun positions(): Collection<ChunkPos>
    fun chunks(): Collection<Chunk>
    fun size(): Int
}
